### PR TITLE
various fixes

### PR DESCRIPTION
**Pull request by trepca**

### CODE REVIEW

File: circuit_puzzles/programs/announcer_configure.clsp [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        cooldown_start
+        (+ current_timestamp delay)
+        input_conditions
+      )
+    )
+    (assert
+      inner_puzzle_hash  ; must be owner
+      (li
+        (assert-statute statutes_puzzle_hash STATUTE_ANNOUNCER_DELAY delay)
+        (assert-statute statutes_puzzle_hash STATUTE_ANNOUNCER_MIN_DEPOSIT deposit)
```
Comment (nimcon) [inline circuit_puzzles/programs/announcer_configure.clsp:71]: do we want deposit to equal Statutes min deposit, or do we want to allow deposit to be any value greater than Statutes min deposit?

File: circuit_puzzles/programs/announcer_configure.clsp [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+            (all
+              APPROVED
+              (> COOLDOWN_START 0)
+              (> (- current_timestamp COOLDOWN_START) deactivation_cooldown_interval)
+            )
+            ; deactivating announcer
+            (c 0 0)
+            (if (= COOLDOWN_START 0)
+              ; starting the cooldown
+              (c 1 (+ deactivation_cooldown_interval current_timestamp))
```
Comment (nimcon) [inline circuit_puzzles/programs/announcer_configure.clsp:32]: shouldn't starting cooldown set COOLDOWN to current_timestamp?

File: circuit_puzzles/programs/announcer_configure.clsp [hunk 3]
```diff
@@ -0,0 +1,88 @@
+; governance backdoor
+; we either activate this announcer or deactivate, can happen at any time via CRT vote
+(mod ((MOD_HASH STATUTES_STRUCT LAUNCHER_ID INNER_PUZZLE_HASH APPROVED DEPOSIT DELAY ATOM_VALUE COOLDOWN_START TIMESTAMP_EXPIRES)
+      ; solution
+      inner_puzzle_hash new_puzzle_hash deposit input_conditions
+      (current_timestamp statutes_inner_puzzle_hash input_toggle_activation deactivation_cooldown_interval delay new_price melt)
```
Comment (nimcon) [inline circuit_puzzles/programs/announcer_configure.clsp:6]: rename new_price to new_atom_value to keep terminology consistent with other files?

File: circuit_puzzles/programs/announcer_configure.clsp [hunk 4]
```diff
@@ -0,0 +1,88 @@
+; governance backdoor
+; we either activate this announcer or deactivate, can happen at any time via CRT vote
```
Comment (nimcon) [inline circuit_puzzles/programs/announcer_configure.clsp:2]: configure is an owner action, not a governance one

File: circuit_puzzles/programs/announcer_configure.clsp (hunk 1 above)
Comment (trepca) [reply circuit_puzzles/programs/announcer_configure.clsp:71]: equal to, need to rename it

File: circuit_puzzles/programs/announcer_configure.clsp (hunk 2 above)
Comment (trepca) [reply circuit_puzzles/programs/announcer_configure.clsp:32]: yeah, that's a bug

File: circuit_puzzles/programs/announcer_configure.clsp [hunk 5]
```diff
@@ -1,7 +1,7 @@
-(mod ((MOD_HASH STATUTES_STRUCT LAUNCHER_ID INNER_PUZZLE_HASH APPROVED DEPOSIT DELAY ATOM_VALUE COOLDOWN_START TIMESTAMP_EXPIRES)
+(mod ((MOD_HASH STATUTES_STRUCT LAUNCHER_ID INNER_PUZZLE_HASH APPROVED DEPOSIT DELAY ATOM_VALUE COOLDOWN_START LAST_PENALTY_INTERVAL TIMESTAMP_EXPIRES)
       ; solution
       inner_puzzle_hash new_puzzle_hash deposit input_conditions
-      (current_timestamp statutes_inner_puzzle_hash input_toggle_activation deactivation_cooldown_interval delay new_atom_value melt)
+      (current_timestamp statutes_inner_puzzle_hash input_toggle_activation deactivation_cooldown_interval min_deposit delay new_atom_value melt)
```
Comment (nimcon) [inline circuit_puzzles/programs/announcer_configure.clsp:4]: rename delay -> max_delay and have an extra solution arg delay in the row above? this way we can check that delay <= max_delay further down, allowing Announcers to reduce their respective announcer's delay prior to a delay reduction by governance. o/w Announcers would need to scramble to reduce their delay in the same block as the enactment or risk getting penalised straight away.

File: circuit_puzzles/programs/oracle_mutation.clsp [hunk 6]
```diff
@@ -7,10 +7,8 @@
                 price_updatable_after_seconds
                 price_updatable_threshold_percent
                 price_expiration_seconds
-                announcer_delay
-                announcer_deposit
                 current_timestamp
-                ; price_announcers -> ((inner_puzzle_hash args_hash price))
+                ; price_announcers -> ((inner_puzzle_hash args_hash price deposit))
```
Comment (nimcon) [inline circuit_puzzles/programs/oracle_mutation.clsp:11]: price_announcers would need to include delay as well if we allow this to be less than the max defined in Statutes

File: circuit_puzzles/programs/announcer_penalize.clsp [hunk 7]
```diff
@@ -14,37 +14,60 @@
...
+    penalized_deposit (if (> current_timestamp TIMESTAMP_EXPIRES)
+      ; if it's expired, we penalize on expiration time
+      (/ (* announcer_min_deposit
+        (*
+          (/ expired_for_minutes penalty_interval_in_minutes)
+          penalty_factor_per_interval)
+        )
+        PERCENT_PRECISION
+      )
+      ; if not expired, we penalize per full deposit
```
Comment (nimcon) [inline circuit_puzzles/programs/announcer_penalize.clsp:27]: too harsh? can get penalised every block

File: circuit_puzzles/programs/oracle_mutation.clsp (hunk 6 above)
Comment (trepca) [reply circuit_puzzles/programs/oracle_mutation.clsp:11]: this is invalid comment

**Review by nimcon — APPROVED** [review]

**Review by nimcon — APPROVED** [review]

**Pull request merged**